filebeat/crawler/prospector.go

```diff
@@ -35,9 +36,42 @@ func (p *Prospector) Init() error {
 		return err
 	}
 
+	var prospectorer Prospectorer
+
+	switch p.ProspectorConfig.Harvester.InputType {
+	case cfg.StdinInputType:
+		prospectorer = ProspectorStdin{
+			Prospector: p,
```

Comment:
I'm just no fan of circular dependencies if not really required. is running and registrar really required in by both?

e.g. not requiring the Run method on Prospectorer, but having `OnEvent` should remove requirement on running. The registrar itself is only required by `ProspectorLog`, but not `ProspectorStdin`.